python/tank/util/shotgun.py

```diff
@@ -166,13 +166,68 @@ def __get_sg_config_data(shotgun_cfg_path, user="default"):
     # validate the config data to ensure all fields are present
     if "host" not in config_data:
         raise TankError("Missing required field 'host' in config '%s'" % shotgun_cfg_path)
-    if "api_script" not in config_data:
-        raise TankError("Missing required field 'api_script' in config '%s'" % shotgun_cfg_path)
-    if "api_key" not in config_data:
-        raise TankError("Missing required field 'api_key' in config '%s'" % shotgun_cfg_path)
-    
+
     return config_data
 
+
+def _is_script_user_authenticated(config_data):
+    """
+    Indicates if we are authenticating with a script user for a given configuration.
+    :param config_data: The configuration data
+    :returns: True is we are using a script user, False otherwise.
+    """
+    return config_data.get("api_script") and config_data.get("api_key")
+
+
+def is_script_user_authenticated():
```

Comment:
Does it make more sense if this method is called `needs_authentication` or similar? It seems to be used for that sort of purpose in the tank_cmd script - to check "are we in always-valid mode or do we need to authenticate?"